### PR TITLE
New helm chart for kanister-operator

### PR DESCRIPTION
**Pull request by depohmel**
Incubator chart: https://kanister.io/
Quick start intrustion: https://github.com/kanisterio/kanister

### CODE REVIEW

File: incubator/kanister-operator/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+The command removes all the Kubernetes components associated with the chart and deletes the release.
+
+## Configuration
+
+The following tables lists the configurable parameters of the nginx-ingress chart and their default values.
+
+Parameter | Description | Default
+--- | --- | ---
+`rbac.create` | all required roles and SA will be created | `true`
+`rback.service_account`| porvided service account name will be used | `None`
```
Comment (paulczar) [inline incubator/kanister-operator/README.md]: should be `rbac.service_account`

**Review by paulczar — CHANGES_REQUESTED** [review]

File: incubator/kanister-operator/README.md (hunk 1 above)
Comment (depohmel) [reply incubator/kanister-operator/README.md]: @paulczar thanks.  /done

**Comment by prydonius** [comment]
@depohmel if the chart meets all the technical requirements and documentation requirements in https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements this can be put straight into the stable repository. Can't see anything obvious violating those requirements.

**Comment by depohmel** [comment]
thanks, @prydonius 
Can I do move as part of this PR or will it be better to create a new one?

**Comment by prydonius** [comment]
@depohmel ideally in this PR, so we can avoid it being in incubator at all.

**Comment by depohmel** [comment]
@prydonius, thanks.  Just pushed a move.

**Comment by tdmanv** [comment]
@prydonius thanks for all the help. Is there anything left we can do to get this merged?

File: stable/kanister-operator/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
+description: Kanister-operator Helm chart for Kubernetes
+name: kanister-operator
+version: 0.2.0
```
Comment (unguiculus) [inline stable/kanister-operator/Chart.yaml:4]: Add `appVersion` and `sources`. Use Github username for maintainers.

**Review by unguiculus — CHANGES_REQUESTED** [review]
In general, please follow our guidelines:
https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Make sure you use namespaced templates. Ideally you use `helm create` with Helm 2.8+ and take the `_helpers.tpl` you get.

Note that our RBAC best practices have been revised. Please update accordingly.

https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

Go over readme and notes again. From quickly skimming over them I noticed a few typos.

**Comment by depohmel** [comment]
@unguiculus , thanks for the review.

File: stable/kanister-operator/templates/rbac.yaml [hunk 3]
```diff
@@ -3,50 +3,52 @@ kind: ClusterRole
...
+{{ include "kanister-operator.helmLabels" . | indent 4 }}
+  name: {{ template "kanister-operator.fullname" . }}-cluster-role
 rules:
 - apiGroups:
   - cr.kanister.io
   resources:
   - "*"
   verbs:
   - "*"
+{{- if .Values.serviceAccount.create }}
```
Comment (unguiculus) [inline stable/kanister-operator/templates/rbac.yaml]: Please put the service account into its own file.

File: stable/kanister-operator/templates/_helpers.tpl [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+*/}}
+{{- define "kanister-operator.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/* Helm required labels */}}
+{{- define "kanister-operator.helmLabels" -}}
+heritage: {{ .Release.Service }}
+release: {{ .Release.Name }}
+chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
```
Comment (unguiculus) [inline stable/kanister-operator/templates/_helpers.tpl]: Use the `chart` template.

File: stable/kanister-operator/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+apiVersion: extensions/v1beta1
```
Comment (unguiculus) [inline stable/kanister-operator/templates/deployment.yaml]: `apps/v1beta2`

File: stable/kanister-operator/README.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+```console
+$ helm install stable/kanister-operator
+```
+
+## Introduction
+
+This chart bootstraps a kanister-operator deployment on a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.
+
+## Prerequisites
+  - Kubernetes 1.7+ with Beta APIs enabled
```
Comment (unguiculus) [inline stable/kanister-operator/README.md]: 1.8+

We officially support the current and previous minor versions of Kubernetes.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Almost there.

**Comment by depohmel** [comment]
@ultimateboy could you please do another look?

**Review by unguiculus — APPROVED** [review]

**Comment by depohmel** [comment]
Thanks @unguiculus!

**Pull request merged**